x/staking/keeper/msg_server.go

```diff
@@ -610,6 +610,14 @@ func (k msgServer) TokenizeShares(goCtx context.Context, msg *types.MsgTokenizeS
 		return nil, err
 	}
 
+	// Check if the delegator has disabled tokenization
+	if disabled, unlockTime := k.IsTokenizeSharesDisabled(ctx, delegatorAddress); disabled {
+		if unlockTime.IsZero() {
```

Comment:
The lock is stored as a key, value map of address → lock expiration time, but if there is no "unlock" in progress, the time is uninitialized (i.e. equal to the zero value).

So this check is identifying whether there is an unlock in progress.

I can add more comments to make it clear. One other option is to return 3 values from this function:
```
locked(bool), unlockInProgress (bool), unlockTime
```
I generally think it's good to try and avoid returning more than 2 values from a function, but feels like it might be justified in this case

Reply:
Ah I think I understand. So if the user has never disabled tokenization, we get the UNLOCKED response. And disabled will be false, so the whole code block 614-619 will be skipped. I previously didn't process the `if` on 613. Lgtm, ty for explaining.

Reply:
New commit uses enum instead